starfish/src/main/java/org/maxandersen/Starfish.java

```diff
@@ -0,0 +1,238 @@
+    String cmd = "git clone "+args[0];
+    String originUrl = args[0];
+    Path directory = Paths.get(clone_path+repo_name);
+    gitClone(directory, originUrl);
+    
+
+
+
+    //Launching Vscode on the Cloned Directory 
+    System.out.println("Launching  IDE Now");
+    runCommand(directory.getParent(), ide,clone_path+repo_name);
+    return 10;
+    
+}
+
+
+//Function to edit configuration and serves for command line starfish config
+public static void editConfig()throws Exception{
+System.out.println("\n------Starfish Configuration Editor------");
+String clone_path="";//Holds path to  destination Where the Repository Must Be Clonned
+    String ide="";  //Holds command for IDE to open upon
+    String userHome = System.getProperty( "user.home" ); //Get User Home Directory: /home/user_name
+
+    File configFile = new File(userHome + "/starfish-config.yml"); //Loading YAML
+    
+    BufferedReader reader= new BufferedReader(new InputStreamReader(System.in));
+    
+    
+    //Reading Configuration File
+    try {
+        if (!configFile.exists()) {
+            //Incase no config file exists
+            System.out.println("\n-No configurations exist for starfish...You'll have to  configure it");
+            System.out.println("\n-New  configuration file will be created at: " + configFile.getAbsolutePath() + "\n");
+            configFile.createNewFile();
+            }
+            
+            
+            //--------------------------First we'll input preferred IDE from user------------------------------
+
+            int id=0;
+            while(id!=1||id!=2||id!=3){
+            System.out.println("\n--------Chose the preferred IDE --------\n 1.for vscode \n 2.for eclipse \n 3.for IntelliJ_IDEA ");
+            
+            id=Integer.parseInt(reader.readLine());
+            
+            if(id==1){ide="code";System.out.println("Selected IDE:VsCode");break;}
+            else
+            if(id==2){ide="eclipse";System.out.println("Selected IDE:Eclipse");break;}
+            else
+            if(id==3){ide="idea";System.out.println("Selected IDE:IntelliJ_IDEA");break;}
+            else
+            System.out.println("--------Invalid Input Try Again--------");
+            }
+
+            //-----------Now we'll get preferred clone path on local file system from user--------------
+            System.out.println("--------Enter preferred Clonning path--------");
+            clone_path=reader.readLine();
+
+            //----------Now we'll write configurations to the YAML FILE------------
+            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+            Config configuration=new Config(ide,clone_path);
+            mapper.writeValue(configFile, configuration); //Writing to YAML File
+            
+        
+        
+    }
+    catch (Exception e) {
+        e.printStackTrace();
+        System.out.print("Press any key to continue . . . ");
```

Comment:
Why not just throw exception and exit ?

Reply:
In that case, the terminal closes and can't see what caused error/exception.
Thus did that to wait for users response before exit.

Reply:
I am removing it now